[fx] PassResult returned from passes

angelayi opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #81269
* __->__ #80532
* #80531

Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

angelayi committed:
[fx] PassResult returned from passes

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

SherlockNoMad commented on torch/fx/passes/infra/pass_manager.py:
```diff
@@ -133,7 +155,7 @@ class PassManager:
     """
 
-    passes: List[Callable[[nn.Module], nn.Module]] = []
+    passes: List[Callable[[nn.Module], PassResult]] = []
```
You mentioned some passes works with nn.Module, could you please share a code pointer for such case. 

I am thinking if we should constrain the PassBase to only work with GraphModule.

SherlockNoMad reviewed (approved):
LGTM, we can resolve "whether Pass should only work with GraphModule" in the latter PRs.

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi commented:
@angelayi has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D37523211).

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]

angelayi committed:
Update on "[fx] PassResult returned from passes"


Differential Revision: [D37523211](https://our.internmc.facebook.com/intern/diff/D37523211)

[ghstack-poisoned]
